### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Victoria-Moiseienko**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
-public class Car {
+public class Car implements Cloneable {
+    private String name;
+    private Engine engine;
+    private String color;
+    private int doorCount;
+    private int issueYear;
 
+    public Car(String name, Engine engine, String color, int doorCount, int issueYear) {
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:11]: you have got some values in your constructor. What next we supposed to do  with them?

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        Engine cloneEngine = engine;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:15]: will you get the copy of of engine or the same engine?

Comment (Victoria-Moiseienko) [reply src/main/java/core/basesyntax/Car.java]: Fixed. Done

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (Victoria-Moiseienko) [reply src/main/java/core/basesyntax/Car.java]: Fixed. Done

File: src/main/java/core/basesyntax/Car.java [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+        this.engine = engine;
+        this.color = color;
+        this.doorCount = doorCount;
+        this.issueYear = issueYear;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        Car cloneCar = new Car(name, engine.clone(), color, doorCount, issueYear);
+        return cloneCar;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:21]:
```suggestion
        return new Car(name, engine.clone(), color, doorCount, issueYear);
```
There is no reason to create local variable if you nogt gonna use it for something else in your method

**Pull request closed without merging**